zwave_js_server/model/notification.py

```diff
@@ -34,11 +64,66 @@ def node_id(self) -> int:
         return self.data["nodeId"]
 
     @property
-    def notification_label(self) -> str:
+    def command_class(self) -> int:
+        """Return command class."""
+        return self.data["ccId"]
+
+    @property
+    def type_(self) -> int:
+        """Return type property."""
+        return self.data["args"]["type"]
+
+    @property
+    def label(self) -> str:
+        """Return label property."""
+        return self.data["args"]["label"]
+
+    @property
+    def event(self) -> int:
+        """Return event property."""
+        return self.data["args"]["event"]
+
+    @property
+    def event_label(self) -> str:
         """Return notification label property."""
-        return self.data["notificationLabel"]
+        return self.data["args"]["eventLabel"]
 
     @property
     def parameters(self) -> Dict[str, Any]:
         """Return installer icon property."""
-        return self.data.get("parameters", {})
+        return self.data["args"].get("parameters", {})
+
+
+class EntryControlNotification:
+    """Model for a Zwave Node's Entry Control CC notification event."""
+
+    def __init__(self, node: "Node", data: EntryControlNotificationDataType) -> None:
+        """Initialize."""
+        self.node = node
+        self.data = data
+
+    @property
+    def node_id(self) -> int:
+        """Return node ID property."""
+        return self.data["nodeId"]
+
+    @property
+    def command_class(self) -> int:
+        """Return command class."""
+        return self.data["ccId"]
+
+    @property
+    def event_type(self) -> int:
+        """Return event type property."""
+        return self.data["args"]["eventType"]
+
+    @property
+    def data_type(self) -> int:
+        """Return data type property."""
+        return self.data["args"]["dataType"]
+
+    @property
+    def event_data(self) -> Optional[str]:
+        """Return event data property."""
+        event_data = self.data["args"].get("eventData")
+        return parse_buffer(event_data) if event_data else cast(str, event_data)
```

Comment:
Why do we cast to string if `event_data` is falsy? Looks like we're potentially casting `None` to string.

Reply:
good catch